File: VB/DXExample.Module.Web/WebSelectFirstNavigationItemController.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Templates
Imports DevExpress.ExpressApp.Web.Templates.ActionContainers

Namespace DXExample.Module.Web
	Public Class WebSelectFirstNavigationItemController
		Inherits SelectFirstNavigationItemControllerBase
		Public Sub New()
		End Sub
		Protected Overrides Function FindNavBarControl() As DevExpress.ExpressApp.Templates.IActionContainer
			If Window.Template IsNot Nothing Then
				For Each actionContainer As IActionContainer In Window.Template.GetContainers()
					If TypeOf actionContainer Is NavigationTabsActionContainer OrElse TypeOf actionContainer Is NavigationBarActionContainer Then
						Return actionContainer
					End If
				Next actionContainer
			End If
			Return Nothing
		End Function
		Protected Overrides Sub SetupNavigationControl(ByVal navigationControl As IActionContainer)
			If TypeOf navigationControl Is NavigationTabsActionContainer Then
				Dim navTabsControl As NavigationTabsActionContainer = CType(navigationControl, NavigationTabsActionContainer)
				AddHandler navTabsControl.ActiveTabChanged, AddressOf navTabsControl_ActiveTabChanged
				navTabsControl.AutoPostBack = True
			ElseIf TypeOf navigationControl Is NavigationBarActionContainer Then
				Dim navBarControl As NavigationBarActionContainer = CType(navigationControl, NavigationBarActionContainer)
				AddHandler navBarControl.ExpandedChanged, AddressOf navBarControl_ExpandedChanged
				navBarControl.AutoPostBack = True
			End If
		End Sub
		Private Sub navTabsControl_ActiveTabChanged(ByVal source As Object, ByVal e As DevExpress.Web.ASPxTabControl.TabControlEventArgs)
			SelectFirstItem(e.Tab.Index)
		End Sub
		Private Sub navBarControl_ExpandedChanged(ByVal source As Object, ByVal e As DevExpress.Web.ASPxNavBar.NavBarGroupEventArgs)
			SelectFirstItem(e.Group.Index)
		End Sub
	End Class
End Namespace

File: CS/DXExample.Module.Web/WebSelectFirstNavigationItemController.cs
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Templates;
using DevExpress.ExpressApp.Web.Templates.ActionContainers;

namespace DXExample.Module.Web {
    public class WebSelectFirstNavigationItemController : SelectFirstNavigationItemControllerBase {
        public WebSelectFirstNavigationItemController() { }
        protected override DevExpress.ExpressApp.Templates.IActionContainer FindNavBarControl() {
            if (Window.Template != null) {
                foreach (IActionContainer actionContainer in Window.Template.GetContainers()) {
                    if (actionContainer is NavigationTabsActionContainer || actionContainer is NavigationBarActionContainer) {
                        return actionContainer;
                    }
                }
            }
            return null;
        }
        protected override void SetupNavigationControl(IActionContainer navigationControl) {
            if (navigationControl is NavigationTabsActionContainer) {
                NavigationTabsActionContainer navTabsControl = (NavigationTabsActionContainer)navigationControl;
                navTabsControl.ActiveTabChanged += new DevExpress.Web.ASPxTabControl.TabControlEventHandler(navTabsControl_ActiveTabChanged);
                navTabsControl.AutoPostBack = true;
            } else if (navigationControl is NavigationBarActionContainer) {
                NavigationBarActionContainer navBarControl = (NavigationBarActionContainer)navigationControl;
                navBarControl.ExpandedChanged += new DevExpress.Web.ASPxNavBar.NavBarGroupEventHandler(navBarControl_ExpandedChanged);
                navBarControl.AutoPostBack = true;
            }
        }
        private void navTabsControl_ActiveTabChanged(object source, DevExpress.Web.ASPxTabControl.TabControlEventArgs e) {
            SelectFirstItem(e.Tab.Index);
        }
        private void navBarControl_ExpandedChanged(object source, DevExpress.Web.ASPxNavBar.NavBarGroupEventArgs e) {
            SelectFirstItem(e.Group.Index);
        }
    }
}
